cfme/tests/openstack/test_host_power_control.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+from utils import testgen
+from cfme.web_ui import Quadicon
+from cfme.infrastructure.host import Host
+from cfme.web_ui import InfoBlock, toolbar
+from cfme.fixtures import pytest_selenium as sel
+from utils.wait import wait_for
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack-infra'], scope='module')
+
+
+def refresh_host():
+    toolbar.select("Configuration", "Provide Nodes", invokes_alert=True)
+
+
+def is_host_on(node):
+    return node.get_detail("Properties", "Power State") == 'on'
+
+
+@pytest.mark.usefixtures("setup_provider_modscope")
+def test_host_power_off(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    quad = my_quads[0]
+    host = Host(name=quad.name)
+    if is_host_on(host):
+        host.power_off()
+        refresh_host()
+    else:
+        pytest.skip("Node is already off")
+    wait_for(lambda: not is_host_on(host), fail_func=lambda: sel.refresh(),
+             delay=15, timeout="10m")
+    assert not is_host_on(host)
+
+
+def test_host_power_on(provider):
+    provider.load_details()
+    sel.click(InfoBlock.element("Relationships", "Nodes"))
+    my_quads = list(Quadicon.all())
+    assert len(my_quads) > 0
+    quad = my_quads[0]
+    host = Host(name=quad.name)
+    if not is_host_on(host):
+        host.power_on()
+        refresh_host()
+    else:
+        pytest.skip("Node is already on")
+    wait_for(lambda: is_host_on(host), fail_func=lambda: sel.refresh(),
+             delay=15, timeout="10m")
```

Comment:
Consider using the toolbar refresh instead of full browser tab refresh if available. Should be able to import toolbar and call `toolbar.refresh` for fail_func (removing lambda usage as @gshefer suggested as well)